File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/adapters/pager/PagerCardInfinityAdapter.java
package com.sup.dev.android.views.adapters.pager;


public abstract class PagerCardInfinityAdapter extends PagerCardAdapter {

    public static int LOOPS_COUNT = 100000;

    @Override
    protected int realPosition(int position) {
        return position % getRealCount();
    }

    @Override
    public int getCount() {
        int realCount = getRealCount();
        return realCount == 1 ? 1 : realCount * LOOPS_COUNT;
    }

    public int getRealCount() {
        return super.getCount();
    }

}

File: DevSupportAndroid/src/main/java/com/sup/dev/android/views/adapters/pager/PagerCardInfinityAdapter.kt
package com.sup.dev.android.views.adapters.pager

abstract class PagerCardInfinityAdapter : PagerCardAdapter() {

    val realCount: Int
        get() = super.getCount()

    override fun realPosition(position: Int): Int {
        return position % realCount
    }

    override fun getCount(): Int {
        val realCount = realCount
        return if (realCount == 1) 1 else realCount * LOOPS_COUNT
    }

    companion object {

        var LOOPS_COUNT = 100000
    }

}
